mantidproject/mantid
# ISIS Powder Pearl Prevent issues in Vanadium Spline

**Description of work.**
Changed generate_and_focus_output so that Pearl create vanadium does not lead to lower than expected values in vandium spline. The rebin in MergeRuns was causing the output workspace to have bins containing only partial data at either end of the vanadium, these gave the sudden drop in values that was effecting the vanadium spline, this is now avoided by cropping the partial bins after the merge.
**To test:**
(requires access to the isis data search archive)
Run the following script with these [files](https://github.com/mantidproject/mantid/files/2486472/PEARL.zip) (additional files may be necessary, but were too large for github)




```
from isis_powder import Pearl

config_file_path = r"/path/to/pearl_shared_config.yaml"
Pearl_example = Pearl(config_file=config_file_path,user_name="testold70")
Pearl_example.create_vanadium(run_in_cycle=103760, do_absorb_corrections=True)
Pearl_example.focus(run_number=103773)#"101508-101515")
```
Plot PRL103759-103767_tt70_mods1-9, if the changes have worked, there should not be a sudden drop at the beginning.
Fixes #23557 
Re #23092


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
